pygfx/renderers/wgpu/_pipelinebuilder.py

```diff
@@ -28,12 +29,28 @@ def ensure_pipeline(renderer, wobject):
     new_pipeline_infos = None
 
     # Do we need to recreate the pipeline_objects?
-    if wobject.rev != wobject_pipeline.get("ref", ()):
+
+    # Check if render state has changed
+    state_hash = ""
+    for key in state:
+        values = state[key]
+        for value in values:
+            assert hasattr(value, "id")
+            state_hash += f"{key}.{value.id}_"
```

Comment:
No wait, a user could render the same scene using different renderers ...

Reply:
In fact, now, ”render state“ is not associated with anything. What I thought was that when each time `renderer.render()` method  was called, it was used to analyze some states (lights, fog, etc.) that affect the global situation in the currently rendered scene, therefore it was emptied and reassembled every time in `renderer.render()`, and used to judge whether the render pipelines need to be updated.

There are some problems here indeed. When the state changes, only the objects affected by the state need to update their render pipeline (currently, only objects with `MeshPhongMaterial` or `MeshFlashMaterial`), rather than all objects in the scene. I need to find some ways to improve it.

Reply:
I thought later that maybe render state should be associated with `scene`, or use the global cache with `hash(scene) `as the key to manage the state